framework/src/main/java/org/apache/drill/test/framework/TestDriver.java

```diff
@@ -264,26 +270,135 @@ public int runTests() throws Exception {
     			memUsage[1][0], memUsage[1][1], memUsage[1][2]));
       }
 
-      totalPassingTest += passingTests.size();
-      totalExecutionFailure += executionFailures.size();
-      totalVerificationFailure += verificationFailures.size();
-      totalTimeoutFailure += timeoutFailures.size();
-      totalCanceledTest += canceledTests.size(); 
+      totalPassingTests += passingTests.size();
+      totalExecutionFailures += executionFailures.size();
+      totalVerificationFailures += verificationFailures.size();
+      totalTimeoutFailures += timeoutFailures.size();
+      totalCanceledTests += canceledTests.size();
+      totalFailingTests = 0;
+      totalFailingTests = totalExecutionFailures + totalVerificationFailures + totalTimeoutFailures + totalCanceledTests;
+      failedTests.addAll(executionFailures);
+      failedTests.addAll(verificationFailures);
+      failedTests.addAll(timeoutFailures);
+      failedTests.addAll(canceledTests);
+ 
     }
+   
+     if(cmdParam.percentThresholdForRandomFailure >= 0.1 ){
+     	cmdParam.percentThresholdForRandomFailure = 0.1;
+     } 
+     if (totalFailingTests/(totalPassingTests +totalFailingTests) <= cmdParam.percentThresholdForRandomFailure && totalFailingTests != 0){
+      LOG.info("Found "+totalFailingTests+".Running the failed tests again to isolate random failures ");
```

Comment:
Change to
LOG.info("Found "+totalFailingTests+" failures. Running the failed tests again to isolate random failures");